### PR TITLE
fix: Python dependencies in #11499

### PR DESCRIPTION
**Pull request by john-bodley**
### SUMMARY

This PR fixes https://github.com/apache/incubator-superset/pull/11499 which requires Redis to be a core Python dependency for the `AsyncQueryManager` Flask extension.

### TEST PLAN

CI. Note the dependencies were updated via `pip-compile-multi --no-upgrade`.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: requirements/base.in [hunk 1]
```diff
@@ -17,4 +17,4 @@
 # under the License.
 -e file:.
 pyrsistent>=0.16.1,<0.17
-importlib-metadata>=1.7.0, <2
```
Comment (john-bodley) [inline requirements/base.in:20]: Needed to resolve conflicting versions.

File: requirements/development.in [hunk 2]
```diff
@@ -17,10 +17,10 @@
 # under the License.
 -r base.in
 flask-cors>=2.0.0
+mysqlclient==1.4.2.post1
+pillow>=7.0.0,<8.0.0
 pydruid>=0.6.1,<0.7
 pyhive[hive]>=0.6.1
-tableschema
```
Comment (john-bodley) [inline requirements/development.in:22]: ABC.

**Review by ktmud — APPROVED** [review]
lgtm

**Comment by nytai** [comment]
`pip-compile-multi --no-upgrade` -- oh, so that's how you add a dependency without upgrading every other one!

File: requirements/testing.txt [hunk 3]
```diff
@@ -35,9 +35,9 @@ pyhive[hive,presto]==0.6.3  # via -r requirements/development.in, -r requirement
 pylint==2.6.0             # via -r requirements/testing.in
 pytest-cov==2.10.1        # via -r requirements/testing.in
 pytest==6.1.2             # via -r requirements/testing.in, pytest-cov
-redis==3.5.3              # via -r requirements/testing.in
```
Comment (robdiciuccio) [inline requirements/testing.txt:38]: Redis is currently required for running tests.

Comment (john-bodley) [reply requirements/testing.txt:38]: @robdiciuccio this will be inherited from `base.in`.

File: requirements/docker.txt [hunk 4]
```diff
@@ -10,7 +10,6 @@
 gevent==20.9.0            # via -r requirements/docker.in
 greenlet==0.4.17          # via gevent
 psycopg2-binary==2.8.6    # via -r requirements/docker.in
-redis==3.5.3              # via -r requirements/docker.in
```
Comment (robdiciuccio) [inline requirements/docker.txt:13]: This is replaced by the requirement in `base`, yes?

**Review by robdiciuccio — APPROVED** [review]

**Pull request merged**